proto/message_contents/signature.proto

```diff
@@ -22,8 +22,16 @@ message Signature {
         bytes bytes = 1; // compact representation [ R || S ], 64 bytes
         uint32 recovery = 2; // recovery bit
     }
+    // Self-describing signature format meant for maximizing interoperability
+    message Generic {
+        bytes signature_bytes = 1; // signature bytes
+        bytes message_bytes = 2;
+        string algorithm = 3; // signature algorithm
```

Comment:
This also feels like it should be an enum. We can always add values as we add more signature algorithms.